x-pack/plugins/lens/public/pie_visualization/toolbar.tsx

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPopover,
+  EuiSelect,
+  EuiButtonEmpty,
+  EuiFormRow,
+  EuiSuperSelect,
+  EuiRange,
+  EuiSwitch,
+  EuiHorizontalRule,
+  EuiSpacer,
+  EuiButtonGroup,
+} from '@elastic/eui';
+import { Position } from '@elastic/charts';
+import { DEFAULT_PERCENT_DECIMALS } from './constants';
+import { PieVisualizationState, SharedLayerState } from './types';
+import { VisualizationToolbarProps } from '../types';
+import './toolbar.scss';
+
+const numberOptions: Array<{ value: SharedLayerState['numberDisplay']; inputDisplay: string }> = [
+  {
+    value: 'hidden',
+    inputDisplay: i18n.translate('xpack.lens.pieChart.hiddenNumbersLabel', {
+      defaultMessage: 'Hide from chart',
+    }),
+  },
+  {
+    value: 'percent',
+    inputDisplay: i18n.translate('xpack.lens.pieChart.showPercentValuesLabel', {
+      defaultMessage: 'Show percent',
+    }),
+  },
+  {
+    value: 'value',
+    inputDisplay: i18n.translate('xpack.lens.pieChart.showFormatterValuesLabel', {
+      defaultMessage: 'Show value',
+    }),
+  },
+];
+
+const categoryOptions: Array<{
+  value: SharedLayerState['categoryDisplay'];
+  inputDisplay: string;
+}> = [
+  {
+    value: 'default',
+    inputDisplay: i18n.translate('xpack.lens.pieChart.showCategoriesLabel', {
+      defaultMessage: 'Inside or outside',
+    }),
+  },
+  {
+    value: 'inside',
+    inputDisplay: i18n.translate('xpack.lens.pieChart.fitInsideOnlyLabel', {
+      defaultMessage: 'Inside only',
+    }),
+  },
+  {
+    value: 'hide',
+    inputDisplay: i18n.translate('xpack.lens.pieChart.categoriesInLegendLabel', {
+      defaultMessage: 'Hide labels',
+    }),
+  },
+];
+
+const categoryOptionsTreemap: Array<{
+  value: SharedLayerState['categoryDisplay'];
+  inputDisplay: string;
+}> = [
+  {
+    value: 'default',
+    inputDisplay: i18n.translate('xpack.lens.pieChart.showTreemapCategoriesLabel', {
+      defaultMessage: 'Show labels',
+    }),
+  },
+  {
+    value: 'hide',
+    inputDisplay: i18n.translate('xpack.lens.pieChart.categoriesInLegendLabel', {
+      defaultMessage: 'Hide labels',
+    }),
+  },
+];
+
+const legendOptions: Array<{
+  value: SharedLayerState['legendDisplay'];
+  label: string;
+  id: string;
+}> = [
+  {
+    id: 'pieLegendDisplay-default',
```

Comment:
complete nitpick:
this is the example legendOption object for pie:
```
{
    id: 'pieLegendDisplay-show',
    value: 'show',
    label: i18n.translate('xpack.lens.pieChart.alwaysShowLegendLabel', {
      defaultMessage: 'show',
    }),
  }
```

and this is the example for xy:
```
 {
    id: `xy_legend_show`,
    value: 'show',
    label: i18n.translate('xpack.lens.xyChart.legendVisibility.show', {
      defaultMessage: 'show',
    }),
  },
```

I think it could be nice to use the same standard with namings here. No biggy though.